framework/src/main/java/org/checkerframework/framework/util/MultiGraphQualifierHierarchy.java

```diff
@@ -91,16 +92,71 @@ public void addQualifier(AnnotationMirror qual) {
                 return;
             }
 
-            Name pqtopclass = QualifierPolymorphism.getPolymorphicQualifierElement(qual);
+            Name pqtopclass = getPolymorphicQualifierElement(qual);
             if (pqtopclass != null) {
-                AnnotationMirror pqtop =
-                        AnnotationBuilder.fromName(atypeFactory.getElementUtils(), pqtopclass);
-                // use given top (which might be PolymorphicQualifier) as key
+                AnnotationMirror pqtop;
+                if (pqtopclass.contentEquals(Annotation.class.getName())) {
+                    // @PolymorphicQualifier with no value defaults to Annotation.class.
+                    // That means there is only one top in the hierarchy. The top qualifier
+                    // may not be known at this point, so use the qualifier itself.
+                    // This changed to top in MultiGraphQualifierHierarchy.addPolyRelations
```

Comment:
The `This changed` isn't clear. Maybe `It will be changed to...`? Or something else?